jupyterlab/shim-maker.js

```diff
@@ -2,86 +2,59 @@
+ * @param sourceFolder (string) - The source folder (defaults to `lib`).
  *
- * @param regex - A regular expresion used to match files - the default
- *  is all `.js` files.
- *
- * @returns A promise that resolves when the file is created.
+ * @returns The code used to export the entire package.
  */
-function shimmer(modName, sourceFolder, regex) {
+function shimmer(modName, sourceFolder) {
+  var dirs = [];
+  var files = [];
+  var lines = [];
+
+  // Find the path to the module.
   var modPath = require.resolve(modName + '/package.json');
   sourceFolder = sourceFolder || 'lib';
-  regex = regex || /.*\.js$/;
   modPath = path.join(path.dirname(modPath), sourceFolder);
-  var lines = ['var ' + modName + ' = {}'];
 
-  var entries = fs.readdirSync(modPath);
-  // Algorithm:
-  // search for index.js file
-  // if one exists, use it, otherwise use an empty object to initialize
-  // then, for each, if it is a directory, recurse-and-add
-  // if it is a file that matches our regex, add it
-  var lines = getLines(modName, sourceFolder, modPath, modPath, regex);
-  lines.push('module.exports = ' + modName);
-  return lines.join('\n');
-}
-
-
-/**
- * Get the appropriate shim lines for the items in a folder.
- */
-function getLines(modName, sourceFolder, basePath, currentPath, regex) {
-  var lines = [];
-  var entries = fs.readdirSync(currentPath);
-  var relPath = path.relative(basePath, currentPath);
-  var parts = relPath.split('/');
-  var modPath = path.join(modName, sourceFolder, relPath);
-
-  // Initialize with the index file if available, else an empty object.
-  if (entries.indexOf('index.js') !== -1) {
-    if (!parts[0]) {
-      lines.push('var ' + modName + ' = require("' + modPath + '");');
-    } else {
-      lines.push(modName + '["' + parts.join('"]["') + '"] = require("' + modPath + '");');
-    }
-  } else {
-    if (!parts[0]) {
-      lines.push('var ' + modName + ' = {};');
-    } else {
-      lines.push(modName + '["' + parts.join('"]["') + '"] = {};');
-    }
-  }
-  // Otherwise, handle each file.
+  // Walk through the source tree.
+  var entries = walkSync.entries(modPath);
   for (var i = 0; i < entries.length; i++) {
+    // Get the relative path to the entry.
     var entry = entries[i];
-    // Handle directories recursively.
-    var isDirectory = fs.statSync(path.join(currentPath, entry)).isDirectory();
-    if (isDirectory) {
-      lines = lines.concat(getLines(modName, sourceFolder, basePath, path.join(currentPath, entry), regex));
-    } else if (regex.test(entry)) {
-      // Skip index files, already taken care of.
-      if (path.basename(entry) === 'index.js') {
-        continue;
-      }
-      entry = entry.replace(path.extname(entry), '');
-      var entryParts = parts.slice();
-      entryParts.push(entry);
-      if (!entryParts[0]) {
-        entryParts = entryParts.slice(1);
+    var entryPath = path.join(entry.basePath, entry.relativePath);
+    entryPath = path.relative(modPath, entryPath);
```

Comment:
This looks like you're just reconstructing `entry.fullPath` (https://github.com/joliss/node-walk-sync/blob/master/index.js#L121). Also, why not just use `entry.relativePath` to begin with, since it seems like `basePath` is `modPath`